### PR TITLE
switch to minimal scaleway SDK

### PR DESCRIPTION
**Pull request by nicolai86**
this PR aims to switch the scaleway provider to the new, minimal scaleway SDK located [here](https://github.com/nicolai86/scaleway-sdk).

This ensures that the SDK doesn't include a huge number of unnecessary dependencies and also allows us to have a more consistent way to interface with Scaleway.

Also it fixes the broken compilation on solaris as a result of the removed dependencies:

```
[10:17:05]    
[10:17:05]    1 errors occurred:
[10:17:05]    --> solaris/amd64 error: exit status 2
[10:17:05]    Stderr: # github.com/terraform-providers/terraform-provider-scaleway/vendor/github.com/moul/gotty-client
[10:17:05]    vendor/github.com/moul/gotty-client/arch.go:25: undefined: syscall.SYS_IOCTL
[10:17:05]    
[10:17:05]    make: *** [compile] Error 1
[10:17:05]    Process exited with code 2
```

- [x] clean up Scaleway SDK (remove code unrelated to API interface)
- [x] remove no longer needed imports

### CODE REVIEW

**Comment by nicolai86** [comment]
@stack72  would love to have a second pair of eyes on this PR. I've slimmed down the SDK and removed no longer necessary dependencies. Besides splitting up the monolithic `api.go` I did not introduce many changes (renaming constructor, removing some unnecessary attributes from the `ScalewayAPI` type).

Once this is merged I can move forward with adding tests to the SDK (there were none before) as well as migrating some helpers from the terraform provider back to the Scaleway SDK.

File: scaleway/data_source_image.go [hunk 1]
```diff
@@ -59,19 +60,36 @@ func scalewayImageAttributes(d *schema.ResourceData, img *api.ScalewayImage) err
...
-	var needle string
+	var nameMatch func(api.MarketImage) bool
 	if name, ok := d.GetOk("name"); ok {
-		needle = name.(string)
+		nameMatch = func(img api.MarketImage) bool {
+			return img.Name == name.(string)
+		}
 	} else if nameFilter, ok := d.GetOk("name_filter"); ok {
-		needle = nameFilter.(string)
+		exp := regexp.MustCompile(nameFilter.(string))
```
Comment (apparentlymart) [inline scaleway/data_source_image.go]: It looks like this would panic if the user provided an invalid regexp for `name_filter`. Bad user input should not cause a panic, since a panic can cause data loss. Would prefer to use `regexp.Compile` and return a good error if it fails.

Comment (nicolai86) [inline scaleway/data_source_image.go]: Good catch!

Comment (nicolai86) [inline scaleway/data_source_image.go]: fixed!

**Comment by nicolai86** [comment]
:heart: @apparentlymart . ty.
